community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/spi/RepeatableReadQueryContext.scala

```diff
@@ -113,11 +114,51 @@ final class RepeatableReadQueryContext(inner: QueryContext, locker: Locker) exte
     locker.acquireLock(relationshipOps.getByInnerId(id))
   }
 
-  private def lockAll[T <: PropertyContainer](iter: Iterator[T]): Iterator[T] = iter.map {
-    item =>
-      locker.acquireLock(item)
-      item
+
+  private def lockAllNodesAndFilter(inner: Iterator[Node])(f: Node => Boolean): Iterator[Node] = new Iterator[Node] {
+    var nextNode: Option[Node] = null
+
+    private def fetchNext() =
+      if (inner.isEmpty) {
+        nextNode = None
+      }
+      else {
+        var found: Option[Node] = None
+        while (inner.hasNext && found.nonEmpty) {
```

Comment:
This `while` is never going to be entered, is it? `found.nonEmpty` will always be false at this point, or am I missing something?